src/Orchard.Recipes.Implementations/Services/RecipeStepExecutor.cs

```diff
@@ -45,13 +45,20 @@ public class RecipeStepExecutor : IRecipeStepExecutor
                 ExecutionId = executionId
             };
 
-            await _eventBus.NotifyAsync<IRecipeEventHandler>(e => e.RecipeStepExecutingAsync(executionId, recipeStep));
+            await _eventBus
+                .NotifyAsync<IRecipeEventHandler>(e => e.RecipeStepExecutingAsync(executionId, recipeStep));
 
-            await _recipeHandlers.InvokeAsync(rh => rh.ExecuteRecipeStepAsync(recipeContext), _logger);
+            IList<Task> tasks = new List<Task>();
+            foreach (var handler in _recipeHandlers)
+            {
+                tasks.Add(handler.ExecuteRecipeStepAsync(recipeContext));
+            }
+            Task.WaitAll(tasks.ToArray());
```

Comment:
You can't do that here, recipe steps have to be run linearly, you can't do parallelism here. Revert this block.